File: Strings/toggleCase.py
def toggle(s):
    l = list(s)
    diff = abs(ord("a") - ord("A"))
    for i in range(len(l)):
        if "a" <= l[i] <= "z":
            l[i] = chr(ord(l[i]) - diff)
        elif "A" <= l[i] <= "Z":
            l[i] = chr(ord(l[i]) + diff)

    return "".join(l)


def toggleXor(s):
    l = list(s)
    for i in range(len(l)):
        l[i] = chr(ord(l[i]) ^ 32)
    return "".join(l)


print(toggle("aBcDEf"))
print(toggleXor("aBcDEf"))

File: Strings/ToggleCase.java

public class ToggleCase {
    static String toggle(String s) {
        StringBuilder sb = new StringBuilder(s);
        int diff = Math.abs('a' - 'A');
        for (int i = 0; i < sb.length(); i++) {
            char c = sb.charAt(i);
            if (c >= 'a' && c <= 'z') {
                sb.setCharAt(i, c -= diff);
            } else if (c >= 'A' && c <= 'Z') {
                sb.setCharAt(i, c += diff);
            }
        }
        return sb.toString();
    }

    static String toggleXor(String s) {
        StringBuilder sb = new StringBuilder(s);
        for (int i = 0; i < sb.length(); i++) {
            char c = sb.charAt(i);
            sb.setCharAt(i, (char) (c ^ 32));
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        String s = "aBcAEd";
        System.out.println(toggle(s));
        System.out.println(toggleXor(s));
    }
}
